src/huggingface_hub/utils/_headers.py

```diff
@@ -125,6 +125,16 @@ def build_hf_headers(
     # Get auth token to send
     token_to_send = get_token_to_send(token)
 
+    # Retrieve user-agent origin headers from environment variable
+    origin = constants.HF_HUB_USER_AGENT_ORIGIN
```

Comment:
To avoid the `if isinstance(...): ... elif: ... else: ...` logic I would move this logic inside `_http_user_agent` directly.